numba/np/ufunc/ufuncbuilder.py

```diff
@@ -82,6 +83,10 @@ def compile(self, sig, locals={}, **targetoptions):
         topt.update(targetoptions)
 
         flags = compiler.Flags()
+        if 'nogil' in topt:
+            warnings.warn("nogil option is not required, ufuncs and gufuncs "
+                          "always releases GIL for CPU target")
```

Comment:
Suggest using `numba.errors.NumbaWarning` as the warning category such that it's easier for users to filter out things from Numba.